end-to-end/src/test/java/nl/tudelft/wdm/group1/endtoend/EndToEndTest.java

```diff
@@ -67,138 +27,72 @@ public void createAndDeleteUser() {
-                put("price", "21");
-            }});
-            add(new HashMap<String, String>() {{
-                put("stock", "50");
-                put("name", "stock_2");
-                put("price", "22");
-            }});
-        }};
-
-        List<UUID> stockIds = new ArrayList<>();
-
-        for (Map<String, String> stock : stocks) {
-
-            Response stockResponse = given().params(stock)
-                    .when().post("/stock").andReturn();
-
-            stockResponse.then().statusCode(200);
-
-            UUID id = UUID.fromString(stockResponse.jsonPath().get("id"));
+    @Test
+    public void createAndCheckoutOrder() throws InterruptedException {
+        List<UUID> users = createUsers();
+        UUID user0 = users.get(0);
+        int credit0 = getUserCredit(user0);
+        Assert.assertEquals(0, credit0);
 
-            await().untilAsserted(() -> given().when().get("/stock/" + id).then().statusCode(200));
+        List<UUID> stocks = createStocks();
+        UUID stockItem0 = stocks.get(0);
+        int stockItemPrice0 = getStockPrice(stockItem0);
+        int stockItemAmount0 = getStockAmount(stockItem0);
+        Assert.assertEquals(20, stockItemPrice0);
+        Assert.assertEquals(30, stockItemAmount0);
+
+        UUID stockItem1 = stocks.get(1);
+        int stockItemPrice1 = getStockPrice(stockItem1);
+        int stockItemAmount1 = getStockAmount(stockItem1);
+        Assert.assertEquals(21, stockItemPrice1);
+        Assert.assertEquals(40, stockItemAmount1);
+
+        UUID stockItem2 = stocks.get(2);
+        int stockItemPrice2 = getStockPrice(stockItem2);
+        int stockItemAmount2 = getStockAmount(stockItem2);
+        Assert.assertEquals(22, stockItemPrice2);
+        Assert.assertEquals(50, stockItemAmount2);
+
+        int startCredit = 1000;
+        addCredit(users.get(0), startCredit);
+        Assert.assertEquals(startCredit, getUserCredit(user0));
+        addCredit(users.get(0), startCredit);
+        Assert.assertEquals(startCredit * 2, getUserCredit(user0));
+
+        UUID order = createOrder(user0);
+        ArrayList<UUID> itemIds;
+
+        itemIds = getOrderItemIds(order);
+        Assert.assertEquals(0, itemIds.size());
+
+        // add 2 items
+        addOrderItem(order, stockItem0);
+        addOrderItem(order, stockItem1);
+        itemIds = getOrderItemIds(order);
+        Assert.assertEquals(2, itemIds.size());
+
+        // remove 1 item
+        deleteOrderItem(order, stockItem1);
+        itemIds = getOrderItemIds(order);
+        Assert.assertEquals(1, itemIds.size());
+
+        // add 1 item
+        addOrderItem(order, stockItem2);
+        itemIds = getOrderItemIds(order);
+        Assert.assertEquals(2, itemIds.size());
+
+        // make the transaction
+        checkoutOrder(order);
+        Thread.sleep(2000);
 
-            stockIds.add(id);
-        }
+        int newCredit = getUserCredit(user0) - stockItemPrice0 - stockItemPrice2;
```

Comment:
This makes zero sense because getUserCredit does a rest call and already contains the new credit

Reply:
ah good catch, It shuld be startCredit * 2 or something like that indeed